typed/helpers.go

```diff
@@ -116,12 +97,12 @@ func resolveSchema(s *schema.Schema, tr schema.TypeRef, v *value.Value, ah atomH
 
 func deduceAtom(a schema.Atom, v *value.Value) schema.Atom {
 	switch {
-	case v == nil:
-	case v.FloatValue != nil, v.IntValue != nil, v.StringValue != nil, v.BooleanValue != nil:
+	case v == nil, *v == nil:
```

Comment:
This function looks to not have handled `v.Null == true` correctly previously. Is dropping down to the `return a` line the right thing to do?

Reply:
For `v.Null == true`, we intentionally dropped down into the `return a` line, since lists and maps can both be null, and if it's an associative list for example, then we didn't want someone applying null to delete all the entries in the list.

Reply:
Whether that's the desired behavior is maybe up for debate. For a concrete example, the way v.Null is treated now (in master) means that these two operations are allowed without conflicts, and if we changed v.Null == true to act like a scalar, then the second operation here fails with a conflict on ".a":
https://github.com/kubernetes-sigs/structured-merge-diff/blob/e9173f465414834d9f259ca30766989a2183a390/merge/multiple_appliers_test.go#L712-L730